### PR TITLE
Update LWC language server

### PR DESCRIPTION
**Pull request by vazexqi**
### What does this PR do?

Updates the LWC language server.

### CODE REVIEW

File: packages/salesforcedx-vscode-lwc/package.json [hunk 1]
```diff
@@ -1,15 +1,15 @@
 {
   "name": "salesforcedx-vscode-lwc",
-  "displayName": "Lightning Code Editor for Visual Studio Code",
-  "description": "Provides code-editing features for Lightning components",
+  "displayName": "LWC Code Editor for Visual Studio Code",
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-lwc/package.json]: @esalman-sfdc - Just FYI. To be more consistent with the other naming, I'm just going to use LWC throughout.

File: packages/salesforcedx-vscode-lwc/package.json [hunk 2]
```diff
@@ -22,10 +22,10 @@
   },
   "categories": ["Languages"],
   "dependencies": {
+    "lwc-language-server": "0.14.2",
+    "vscode-languageclient": "3.5.0",
     "eslint": "4.15.0",
-    "eslint-plugin-lwc": "0.2.0",
-    "lwc-language-server": "0.14.1",
-    "vscode-languageclient": "3.5.0"
+    "eslint-plugin-lwc": "0.2.4"
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-lwc/package.json]: @rsalvador - Should I use the latest eslint-plugin-lwc with your changes here?

Comment (rsalvador) [reply packages/salesforcedx-vscode-lwc/package.json]: Yes, use version 0.3.2

File: packages/salesforcedx-vscode-lwc/package.json [hunk 3]
```diff
@@ -22,10 +22,10 @@
   },
   "categories": ["Languages"],
   "dependencies": {
+    "lwc-language-server": "0.14.2",
+    "vscode-languageclient": "3.5.0",
     "eslint": "4.15.0",
```
Comment (rsalvador) [inline packages/salesforcedx-vscode-lwc/package.json]: latest eslint is 4.16.0

**Pull request merged**